source/common/upstream/cds_api_impl.cc

```diff
@@ -48,6 +48,13 @@ CdsApiImpl::CdsApiImpl(const envoy::api::v2::core::ConfigSource& cds_config,
 void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::string& version_info) {
   cm_.adsMux().pause(Config::TypeUrl::get().ClusterLoadAssignment);
   Cleanup eds_resume([this] { cm_.adsMux().resume(Config::TypeUrl::get().ClusterLoadAssignment); });
+
+  std::unordered_set<std::string> cluster_names;
+  for (const auto& cluster : resources) {
+    if (!cluster_names.insert(cluster.name()).second) {
+      throw EnvoyException("duplicate clusters found");
```

Comment:
Can you say which cluster is duplicated?

Reply:
I can, if we have multiple need to maintain the list and combine all of them as single string display? -NBD, I can do that.

Reply:
I would just keep it simple and fail on the first one you find and print the cluster...